packages/legend-extension-dsl-mastery/src/graph-manager/protocol/pure/DSL_Mastery_PureProtocolProcessorPlugin.ts

```diff
@@ -257,6 +269,41 @@ export class DSL_Mastery_PureProtocolProcessorPlugin
           V1_buildDataProvider(elementProtocol, context);
         },
       }),
+      new V1_ElementBuilder<V1_MasteryRuntime>({
+        elementClassName: MASTERY_RUNTIME_ELEMENT_CLASS_NAME,
+        _class: V1_MasteryRuntime,
+        firstPass: (
+          elementProtocol: V1_PackageableElement,
+          context: V1_GraphBuilderContext,
+        ): PackageableElement => {
+          const extraMasteryRuntimeBuilders =
+            context.extensions.plugins.flatMap(
+              (plugin) =>
+                (
+                  plugin as DSL_Mastery_PureProtocolProcessorPlugin_Extension
+                ).V1_getExtraMasteryRuntimeFirstPassBuilders?.() ?? [],
+            );
+          for (const builder of extraMasteryRuntimeBuilders) {
+            const metamodel = builder(
+              elementProtocol as V1_MasteryRuntime,
+              context,
+            );
+            if (metamodel) {
+              return metamodel;
+            }
+          }
+          throw new UnsupportedOperationError(
```

Comment:
@gayathrir11 I tried to add a `return undefined` here but was getting Lint Issues as the return type for this function is `PackageableElement` 
Can you please suggest the workaround?

Reply:
The error you are throwing here should be fine as we expect to return a metamodel if we find any protocol of type `V1_Masteryruntime`

Reply:
Just wondering, what about non Alloy implementations of Legend Studio. Will it be necessary for a user to implement a protocol of type V1_MasteryRuntime? Is that fine?